scripts/init.d/cgconfig.in

```diff
@@ -30,37 +30,54 @@
 ### END INIT INFO
 
 # get correct location of binaries from configure
-prefix=@prefix@;exec_prefix=@exec_prefix@;sbindir=@sbindir@
+sbindir=@sbindir@
 CGCONFIGPARSER_BIN=$sbindir/cgconfigparser
 CONFIG_FILE=/etc/cgconfig.conf
+CONFIG_DIR=/etc/cgconfig.d
 servicename=cgconfig
-lockfile=/var/lock/subsys/$servicename
 
+
+lockfile=/run/subsys/$servicename
 #
 # Source LSB routines
 #
-. /lib/lsb/init-functions
+SYSLIBFILE=/lib/lsb/init-functions
+OLDSYSLIBFILE=/etc/init.d/functions
+if [[ -x  $SYSLIBFILE ]] ; then
+  source $SYSLIBFILE
+elif  [[ -x $OLDSYSLIBFILE ]] ; then
+  source $OLDSYSLIBFILE
+  log_warning_msg() ( warning $@ ; printf "\n" 1>&2 ; )
+  log_failure_msg() ( failure $@ ; printf "\n" 1>&2 ; )
+  log_success_msg() ( success $@ ; printf "\n" 1>&2 ; )
+else
+  log_warning_msg() ( printf "warning:%s\n" "$@" 1>&2 ;)
+  log_failure_msg() ( printf "failure:%s\n" "$@" 1>&2 ;)
+  log_success_msg() ( printf "success:%s\n" "$@" 1>&2 ;)
+fi
 
 # read the config
 CREATE_DEFAULT=yes
 if [ -e /etc/sysconfig/cgconfig ]; then
-        . /etc/sysconfig/cgconfig
+        source /etc/sysconfig/cgconfig
 fi
 
+lockfiledir=$(dirname "$lockfile")
+
```

Comment:
`lockfiledir` can be moved below `lockfile` assignment.

Reply:
I don't think so. Note the `source /etc/sysconfig/cgconfig`. The user may override the `$lockfile` in  /etc/sysconfig/cgconfig, and if she does, we would have inconsistent `$lockfile` and `$lockfiledir`.

Reply:
Agree, the user might try to override it but I doubt it's recommended to do so. I do not have any strong opinion here, so let's not move the `lockfiledir`